src/harness/testcases/WINNF_FT_S_EXZ_testcase.py

```diff
@@ -152,87 +98,100 @@ def test_WINNF_FT_S_EXZ_1(self,config_filename):
-    self._sas_admin.InjectExZone({'record': exz_record})
-
     config = loadConfig(config_filename)
     # Very light checking of the config file.
-    self.assertEqual(
-        len(config['registrationRequests']),
-        len(config['expectedResponseCodes']))
+    self.assertNotEqual(len(config['exclusionZoneRecords']) , 0, "Exclusion zone records are zero !!!")
 
-    # Inject FCC IDs.
-    for fcc_id in config['fccIds']:
-      self._sas_admin.InjectFccId({'fccId': fcc_id})
-
-    # Inject user IDs.
-    for user_id in config['userIds']:
-      self._sas_admin.InjectUserId({'userId': user_id})
-   
-    # Register N2, N3 and N4 CBSDs.
-    device_cert = self.getCertFilename('dp_client.cert')
-    device_key = self.getCertFilename('dp_client.key')
+    # Get exclusion zone records from config file.
+    exz_record = {'exclusionZone': config['exclusionZoneRecords']}
 
-    request = {'registrationRequest': config['registrationRequests']}
-    responses = self._sas.Registration(request, device_cert, device_key)['registrationResponse']
+    for i in range(len(exz_record)):
+       # Inject EXZ database records
+       self._sas_admin.InjectExclusionZone({'record': exz_record})
 
-    # Check registration responses.
-    self.assertEqual(len(responses), len(config['registrationRequests']))
-    cbsd_ids = []
-    for i in range(len(responses)):
-      response = responses[i]
-      expected_response_codes = config['expectedResponseCodes'][i]
-      logging.debug('Looking at response number %d', i)
-      logging.debug('Expecting to see response code in set %s in response: %s',
-                    expected_response_codes, response)
-      self.assertIn(response['response']['responseCode'],
-                    expected_response_codes)
-      if response['response']['responseCode'] == 0:  # SUCCESS
-        self.assertTrue('cbsdId' in response)
-        cbsd_ids.append(response[i]['cbsdId'])
-      else:
-        self.assertFalse('cbsdId' in response)
+    # Register the devices
+    cbsd_ids = self.assertRegistered(config['registrationRequests'])
 
-    # Trigger daily activities
+    # Execute CPAS and wait untill completion
     self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+    
+    # Send multiple grant requests
+    grant_request = []
+    for cbsd_id in cbsd_ids:
+      grant = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+      grant['cbsdId'] = cbsd_id
+      #  Update frequency range to partially overlapping with the frequency 
+      #     range of the nearest exclusion zone.
+      grant['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3625000000.0
+      grant['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3635000000.0
+      grant_request.append(grant)
+
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+    #validate the response
+    self.assertEqual(len(response), 8)
+
```

Comment:
You're not actually checking the response codes, as the spec requires you to do.

Reply:
Changes   done to validate N2, N3 and N4 grant responses